File: backend/src/main/java/com/smartcampus/backend/dto/BatimentDTO.java
package com.smartcampus.backend.dto;

import lombok.*;

import java.time.LocalDateTime;
import java.util.List;
// Partie statique  des batiments 
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class BatimentDTO {
    private Long id;
    private String name;
    private String appartenance;
    private String fillColor;
    private String strokeColor;
    private Double centerLat;
    private Double centerLng;
    private Double perimeterMeters;
    private Long campusId;
    private List<List<double[]>> polygonCoordinates;
    private LocalDateTime importedAt;
}

File: frontend/app/src/main/java/com/ObservatoireCampus/mobile/model/Batimentdto.kt
package com.ObservatoireCampus.mobile.model


data class BatimentDto(
    val id: Long,
    val name: String,
    val appartenance: String?,
    val fillColor: String?,
    val strokeColor: String?,
    val campusId: Long?,
    // 3 niveaux : Liste de contours -> Liste de points -> [longitude, latitude]
    val polygonCoordinates: List<List<List<Double>>> = emptyList(),
    val importedAt: String?
)